GooglePlayInstant/Editor/QuickDeploy/GcpClient.cs

```diff
@@ -0,0 +1,297 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using UnityEngine;
+
+namespace GooglePlayInstant.Editor.QuickDeploy
+{
+    /// <summary>
+    /// Provides methods for interacting with Google Cloud Platform (GCP), e.g. to upload an Asset Bundle file to
+    /// Google Cloud Storage.
+    /// </summary>
+    public static class GcpClient
+    {
+        /// <summary>
+        /// Executes all the steps required for deploying a file to GCP according to developer's configuration.
+        /// First verifies if configured bucket exists, and creates the bucket if it does not exist. It then uploads
+        /// the file to GCP and sets the visibility of the file to public.
+        /// </summary>
+        public static void DeployConfiguredFile()
+        {
+            InvokeAccessRestrictedAction(CheckBucketExistence, bucketInfoResponse =>
+            {
+                if (!string.IsNullOrEmpty(bucketInfoResponse.error))
+                {
+                    if (bucketInfoResponse.error.StartsWith("404"))
+                    {
+                        InvokeAccessRestrictedAction(CreateBucket, bucketCreationResponse =>
+                        {
+                            var error = bucketCreationResponse.error;
+                            if (!string.IsNullOrEmpty(error))
+                            {
+                                throw new Exception(string.Format("Got error attempting to create bucket: {0}\n{1}",
+                                    error,
+                                    bucketCreationResponse.text));
+                            }
+
+                            Debug.Log("Google Cloud Storage bucket was successfully created.");
+                            UploadBundleAndMakePublic();
+                        });
+                    }
+                    else
+                    {
+                        throw new Exception(string.Format(
+                            "Got error when verifying bucket existence: {0} \n {1}", bucketInfoResponse.error,
+                            bucketInfoResponse.error));
+                    }
+                }
+                else
+                {
+                    var response = JsonUtility.FromJson<BucketInfoResponse>(bucketInfoResponse.text);
+                    var bucketName = response.name;
+                    if (!string.Equals(bucketName, QuickDeployConfig.Config.cloudStorageBucketName))
```

Comment:
I may have done this wrong in my previous comment, but here you can just do `bucketName != QuickDeployConfig.Config.cloudStorageBucketName` since this isn't Java.